rusk/src/lib/http/chain/graphql.rs

```diff
@@ -199,4 +200,49 @@ impl Query {
             _ => Err(FieldError::new("Specify height or hash")),
         }
     }
+
+    /// Check if a given block height matches a given block hash
+    ///
+    /// If `only_finalized` is set to `true`, only finalized blocks will be
+    /// checked `only_finalized` is set to `false` by default and only works
+    /// on nodes with the archive feature enabled
+    async fn check_block(
+        &self,
+        ctx: &Context<'_>,
+        height: u64,
+        hash: String,
+        only_finalized: Option<bool>,
+    ) -> FieldResult<bool> {
+        if only_finalized.unwrap_or(false) {
+            #[cfg(not(feature = "archive"))]
```

Comment:
This function can go multiple ways but I'm not sure which is correct, couple of points
1. This function doesn't exist when the archive feature is not available
2. If it can still exist and we need to throw an error document the error with a `# Error` section above the function in rust doc